src/psyclone/psyir/transformations/inline_trans.py

```diff
@@ -515,45 +524,81 @@ def _find_routine(call_node):
+                f"unresolved or imported.")
 
     @staticmethod
-    def _find_routine_in_container(
-            call_node, container_symbol):
-        ''' xxx '''
+    def _find_routine_in_container(call_node, container_symbol):
+        '''Searches for the definition of a routine that is being called by
+        the supplied Call. This routine must exist within a container
+        specified by the supplied container symbol.
+
+        :param call_node: the Call that is to be inlined.
+        :type call_node: :py:class:`psyclone.psyir.nodes.Call`
+
+        :param container_symbol: the symbol of the container to search.
+        :type container_symbol: \
+            :py:class:`psyclone.psyir.symbols.ContainerSymbol`
+
+        :returns: the PSyIR for the target routine.
+        :rtype: Optional[:py:class:`psyclone.psyir.nodes.Routine`]
+
+        '''
         # The required Routine will exist within a Container and
         # that Container could exist in the PSyIR as a child of a
         # FileContainer (if the PSyIR contains a
         # FileContainer). If the PSyIR does contain a
         # FileContainer, it will be the root node of the PSyIR.
-        #
-        # The PSyIR does not currently capture whether a Routine
-        # within a Container is private or not. The assumption
-        # here is that a matching Routine will not be private.
-        routine_sym = call_node.routine
+        call_routine_sym = call_node.routine
         for container in call_node.root.children:
             if (isinstance(container, Container) and
                     container.name == container_symbol.name):
                 for routine in container.children:
                     if (isinstance(routine, Routine) and
-                            routine.name == routine_sym.name):
-                        return routine
-                # The Container has been found but it does not
-                # contain the expected Routine. This could be
-                # because the Routine is imported from another
-                # Container. However, searching for a Routine via
-                # multiple Containers is not supported by this
-                # transformation, so stop the search.
+                            routine.name == call_routine_sym.name):
+                        # Check this routine is public
+                        routine_sym = container.symbol_table.lookup(
+                            routine.name)
+                        if routine_sym.visibility == Symbol.Visibility.PUBLIC:
+                            return routine
+                # The Container has been found but it does not contain
+                # the expected Routine or the Routine is not public.
+
+                # Look in the import that names the routine if there is one.
+                table = container.symbol_table
+                for symbol in table.symbols:
+                    if symbol.name == call_routine_sym.name and \
```

Comment:
You can just do `routine_sym = table.lookup(routine.name)` here as you do at L578?

Reply:
Updated. Seems to work :-)